src/services/etherspot.js

```diff
@@ -1026,37 +1027,69 @@ export class EtherspotService {
     const value = EthersUtils.parseUnits(fromValue.toString(), fromAsset.decimals);
 
     try {
-      const quotes = await sdk.getCrossChainQuotes({
+      const routes = await sdk.getAdvanceRoutesLiFi({
         fromTokenAddress: fromAsset.address,
         fromChainId: mapChainToChainId(fromAsset.chain),
         toTokenAddress: toAsset.address,
         toChainId: mapChainToChainId(toAsset.chain),
         fromAmount: value,
       });
 
-      const quote: any = quotes.items[0];
+      if (isEmpty(routes.items)) return null;
 
-      logBreadcrumb('buildCrossChainBridgeTransaction!', 'cross chain bridge quotes', { quotes });
+      const bestRoute = routes.items.reduce((best: Route, route) => {
+        if (!best?.toAmount || EthersBigNumber.from(best.toAmount).lt(route.toAmount)) return route;
+        return best;
+      });
 
-      if (!quote) return null;
+      logBreadcrumb('buildCrossChainBridgeTransaction!', 'cross chain bridge routes', { routes });
 
-      if (!quote?.approvalData) return { transactionData: quote.transaction, quote };
+      if (!bestRoute) return null;
 
-      const tokenAddres = quote.estimate.data.fromToken.address;
-      const { approvalAddress, amount } = quote.approvalData;
+      const { items: advanceRoutesTransactions } = await sdk.getStepTransaction({ route: bestRoute });
 
-      const erc20Contract: any = this.getContract<?EtherspotErc20Interface>(
-        fromAsset.chain,
-        ERC20_CONTRACT_ABI,
-        tokenAddres,
-      );
+      if (isEmpty(advanceRoutesTransactions)) return null;
 
-      if (!erc20Contract) return { transactionData: quote.transaction, quote };
+      let transactions = advanceRoutesTransactions.map((transaction) => {
+        return {
+          from: sdk.state.accountAddress,
```

Comment:
Pleae don't access the account address like this. This value changes depending on the state of the SDK (like if it's not initialised due to an error or race condition, the address is not the smart wallet address). Use the result from `computeContractAccount`

Reply:
Updated it!

Reply:
Thanks!